src/azure-cli/azure/cli/command_modules/appservice/appservice_environment.py

```diff
@@ -3,33 +3,40 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
-from knack.log import get_logger
-from knack.util import CLIError
+# Management Clients
+from azure.mgmt.network import NetworkManagementClient
+from azure.mgmt.web import WebSiteManagementClient
+from azure.mgmt.resource import ResourceManagementClient
 
+# Models
 from azure.mgmt.network.models import (RouteTable, Route, NetworkSecurityGroup, SecurityRule)
-from azure.mgmt.resource.resources.models import DeploymentProperties
+from azure.mgmt.resource.resources.models import (DeploymentProperties, Deployment)
 
+# Utils
 from azure.cli.core.commands import LongRunningOperation
-from azure.cli.core.commands.client_factory import get_mgmt_service_client
+from azure.cli.core.commands.client_factory import (get_mgmt_service_client, get_subscription_id)
 from azure.cli.core.commands.arm import ArmTemplateBuilder
-from azure.cli.core.util import sdk_no_wait
-
-from azure.cli.command_modules.network._client_factory import network_client_factory
+from azure.cli.core.util import (sdk_no_wait, random_string)
+from knack.log import get_logger
+from knack.util import CLIError
+from msrestazure.tools import (parse_resource_id, is_valid_resource_id, resource_id)
 
 VERSION_2019_02_01 = "2019-02-01"
+VERSION_2019_10_01 = "2019-10-01"
+VERSION_2020_04_01 = "2020-04-01"
 
 logger = get_logger(__name__)
 
 
 def list_appserviceenvironments(cmd, resource_group_name=None):
-    ase_client = _get_ase_client_factory(cmd.cli_ctx, VERSION_2019_02_01)
+    ase_client = _get_ase_client_factory(cmd.cli_ctx)
     if resource_group_name is None:
         return ase_client.list()
     return ase_client.list_by_resource_group(resource_group_name)
 
 
 def show_appserviceenvironment(cmd, name, resource_group_name=None):
-    ase_client = _get_ase_client_factory(cmd.cli_ctx, VERSION_2019_02_01)
```

Comment:
I don't think the version here makes any difference AFAIK the version gets set from the shared.py, currently we will always use the default api-version, whichever is the latest. Have you checked if changing this to 2018-02-01 here actually makes a difference. See this PR where I was going to force ASE commands use 2018-02-01 https://github.com/Azure/azure-cli/pull/13737

Reply:
You are right - for the WebServiceManagementClient it seems to fix the version to 2019-08-01. The other ManagementClient adapt to the set api_version. I would be nice to get this fixed, but for now this change with work with both the changed ASE SDK and the changed ARM SDK.

Reply:
Your change does fix the issue, all I am asking is to remove setting the version here since it does nothing.

Reply:
You could just have said so :-) - done.

Reply:
Thanks @madsd